Index: rand_source.c
===================================================================
diff --git a/trunk/third_party/libsrtp/srtp/crypto/rng/rand_source.c b/trunk/third_party/libsrtp/srtp/crypto/rng/rand_source.c
--- a/trunk/third_party/libsrtp/srtp/crypto/rng/rand_source.c	(revision 214783)
+++ b/trunk/third_party/libsrtp/srtp/crypto/rng/rand_source.c	(working copy)
@@ -90,7 +90,7 @@
   dev_random_fdes = RAND_SOURCE_READY;
 #else
   /* no random source available; let the user know */
-  fprintf(stderr, "WARNING: no real random source present!\n");
+  //fprintf(stderr, "WARNING: no real random source present!\n");
   dev_random_fdes = RAND_SOURCE_READY;
 #endif
   return err_status_ok;
